frontend/apps/web/components/site-header/RecordsProgressBar.tsx

```diff
@@ -12,13 +12,13 @@ import { dateToNeoDate, periodToDateRange, UsagePeriod } from '../usage/util';
 
 interface Props {
   identifier: string;
+  idtype: string;
```

Comment:
Let's make this type safe.

here is an example of how we extract the case for the different connections.

```
// Helper function to extract the 'case' property from a config type
type ExtractCase<T> = T extends { case: infer U } ? U : never;

// Extraction type that pulls out all of the connection config cases
export type ConnectionConfigCase = NonNullable<
  ExtractCase<ConnectionConfig['config']>
>;
```

The same can be done for the metric identifiers.